### PR TITLE
Add conversion from Arrow format to Airbyte format in do_put.

### PR DESCRIPTION
**Pull request by revit13**
This PR changes the do_put function to accept tables in Arrow format instead of passing the data in Airbyte format.
The conversion from Arrow format to Airbyte format is done in do_put function.

### CODE REVIEW

File: abm/server.py [hunk 1]
```diff
@@ -175,8 +180,12 @@ def do_put(self, context, descriptor, reader, writer):
             record_reader = reader.to_reader()
             while True:
                 try:
+                  # convert from Arrow format to Airbyte format
                   batch = record_reader.read_next_batch()
-                  connector.write_dataset_bytes(socket, batch.to_pandas().to_json(orient='records', lines=True).encode())
+                  df = batch.to_pandas()
+                  for _, row in df.iterrows():
+                    record_message=AirbyteMessage(type=MessageType.RECORD, record=AirbyteRecordMessage(stream=stream_name, data=row, emitted_at=111)).json()
+                    connector.write_dataset_bytes(socket,record_message.encode('utf-8'))
```
Comment (Mohammad-nassar10) [inline abm/server.py:188]: What is the meaning of the value 111?

Comment (revit13) [reply abm/server.py:188]: its a dummy value. I will change that to take the real time. Thanks

File: abm/server.py [hunk 2]
```diff
@@ -11,9 +11,14 @@
 import json
 import json as simplejson
 import os
+import time
 import socketserver
 from http import HTTPStatus
+import pandas as pd
 import pyarrow.flight as fl
+from airbyte_cdk.models import AirbyteMessage, AirbyteRecordMessage,ConfiguredAirbyteStream,DestinationSyncMode,SyncMode,AirbyteStream,ConfiguredAirbyteCatalog
```
Comment (Mohammad-nassar10) [inline abm/server.py:19]: Please add spaces after the commas.

File: abm/server.py [hunk 3]
```diff
@@ -169,14 +176,22 @@ def do_put(self, context, descriptor, reader, writer):
         with Config(self.config_path) as config:
             asset_conf = config.for_asset(asset_name)
             connector = GenericConnector(asset_conf, logger, self.workdir)
-            command, catalog = connector.create_write_command(schema)
+            # TODO: Params to the Airbyte objects, such as destination_sync_mode, can be configurable using the arrow-flight request
+            streams=[ConfiguredAirbyteStream(destination_sync_mode=DestinationSyncMode.append,sync_mode=SyncMode.full_refresh,
+               stream=AirbyteStream(name=stream_name,supported_sync_modes=[SyncMode.full_refresh],json_schema=json_schema))]
```
Comment (Mohammad-nassar10) [inline abm/server.py:181]: I think this need to be refactored.

File: abm/server.py [hunk 4]
```diff
@@ -157,7 +162,9 @@ def do_put(self, context, descriptor, reader, writer):
         try:
             command = json.loads(descriptor.command)
             asset_name = command['asset']
-            schema = command['schema']
+            stream_name = command['stream_name']
```
Comment (revit13) [inline abm/server.py:165]: auto generate stream_name

File: abm/server.py [hunk 5]
```diff
@@ -169,14 +176,22 @@ def do_put(self, context, descriptor, reader, writer):
...
             connector = GenericConnector(asset_conf, logger, self.workdir)
-            command, catalog = connector.create_write_command(schema)
+            # TODO: Params to the Airbyte objects, such as destination_sync_mode, can be configurable using the arrow-flight request
+            streams=[ConfiguredAirbyteStream(destination_sync_mode=DestinationSyncMode.append,sync_mode=SyncMode.full_refresh,
+               stream=AirbyteStream(name=stream_name,supported_sync_modes=[SyncMode.full_refresh],json_schema=json_schema))]
+            command, catalog = connector.create_write_command(ConfiguredAirbyteCatalog(streams=streams).json())
             socket, container = connector.open_socket_to_container(command)
             idx = 0
             record_reader = reader.to_reader()
+            current_time = time.time() * 1000
```
Comment (revit13) [inline abm/server.py:186]: Check if 1000 is needed

File: sample/sample_put.py [hunk 6]
```diff
@@ -25,41 +46,14 @@ def get_token(self):
...
-                        }, \
-                        "supported_sync_modes": [ \
-                                "full_refresh" \
-                        ] \
-                } \
-            }] \
-        }'
-}
+    "stream_name": "testing",
+    "json_schema": '{"type": "object",  "properties": {"Name": { "type": "string" }}}'
```
Comment (revit13) [inline sample/sample_put.py:50]: Add another column

File: abm/server.py (hunk 3 above)
Comment (revit13) [reply abm/server.py]: @Mohammad-nassar10 I refactored the code. Thanks

File: sample/sample_put.py (hunk 6 above)
Comment (revit13) [reply sample/sample_put.py]: @cdoron I added an additional column.

File: abm/server.py (hunk 5 above)
Comment (revit13) [reply abm/server.py:190]: it is used in airbye code - https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/config_observation.py#L73

**Review by Mohammad-nassar10 — APPROVED** [review]

**Review by cdoron — APPROVED** [review]

**Pull request merged**